### DISCUSSION TITLE
Charge Time; Charge On Value; Charge times

### DISCUSSION THREAD

**Discussion by markuk98** [BottlecapDave/HomeAssistant-OctopusEnergy#660] [Ideas]
Good evening. It would be really nice to add the ability to select how many hours you would like to charge for (ie 3) and then provide the lowest cost to achieve  this (ie the 6th lowest value of the next 61 entries). Also return the times this value (and any values below 0) back as a list. I have achieved the first two by adding the 61 entries to a list, sorting it and selecting the 6th (hours * 2) value from the list. I have not been successful at achieving the last list of charge times. This allows me to charge the car, batteries and heaters for a minimum of 3 hours a day when the _current_rate =< ChargeOnValue. Thank you. PS I used your signup code. 👍 

      - name: "AR_ChargeOnValue" #change this to what you want
        value_template: "{% set ar_chargetable = [
            states('sensor.ar_01') | float(),
            .. some more ...
            states('sensor.ar_61') | float(),
            ] | sort %}                    
            {{ ar_chargetable[states('input_number.ar_charge_hours')|int() *2 ] /100 }}"
(PS. I need to round hours up to the nearest number.)

**Comment by ragg987** [comment]
Hi, I have just been exploring the "Target rate sensor" capability of this integration - I think this is what you are looking for.
https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/target_rate/

**Comment by BottlecapDave** [comment]
Hello and sorry for the late response. @ragg987 is correct in saying that the target rate sensors are probably what you're after based on what you've been describing. It doesn't look at the full data available but only looks at 24 hours at a time to ensure the sensor comes on at least once a day.
